clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -820,39 +821,51 @@ private synchronized void drainNewCalls() {
          * Choose nodes for the calls in the pendingCalls list.
          *
          * @param now           The current time in milliseconds.
-         * @param pendingIter   An iterator yielding pending calls.
+         * @return              The minimum time until a call is ready to be retried if any of the pending
+         *                      calls are backing off after a failure
          */
-        private long chooseNodesForPendingCalls(long now, Iterator<Call> pendingIter) {
+        private long checkPendingCalls(long now) {
```

Comment:
Why did we rename this method? The javadoc and logging seems to indicate that we are still trying to choose nodes for the pending calls?

Reply:
I renamed it for consistency with the singular `checkPendingCall`. I could have named that method `chooseNodeForPendingCall` instead, but I found the return argument not very intuitive since it would not necessarily indicate that a node was successfully chosen, but only that the call had been handled and should be removed from `pendingCalls`. I was not too thrilled with either naming, so happy to hear suggestions.

Reply:
Could `checkPendingCalls` be `maybeChooseNodeForPendingCall`? A bit of a mouthful, but having a `checkX` method that does much more than checking is a bit confusing.

Reply:
Yeah it seems that `checkPendingCalls` could be a bit misleading because the name suggests this is a stateless method. But the method actually changes the state of KafkaAdminClient.